data-protocols/dsp/dsp-http-core/src/main/java/org/eclipse/edc/protocol/dsp/dispatcher/DspHttpRemoteMessageDispatcherImpl.java

```diff
@@ -77,9 +75,10 @@ public <T, M extends RemoteMessage> CompletableFuture<T> send(Class<T> responseT
 
         var request = delegate.buildRequest(message);
 
-        var tokenParameters = TokenParameters.Builder.newInstance()
-                .audience(message.getCounterPartyAddress())
+        var tokenParameters = tokenDecorator.decorate(TokenParameters.Builder.newInstance())
+                .audience(message.getCounterPartyAddress()) // enforce the audience, ignore anything a decorator might have set
```

Comment:
should the aud be the counter-party address?

Reply:
It should be the [recipient of the message](https://www.rfc-editor.org/rfc/rfc7519#section-4.1.3) in some form or shape. The way I read this, we can simply define whether that is the `counterPartyAddress` or something else. We don't have any other piece of "counterparty" information at this point in the flow.